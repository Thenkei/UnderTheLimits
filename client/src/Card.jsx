import React, { Component } from 'react';
import './App.css';
import {
    Col
} from 'react-bootstrap';

class Card extends Component {

  render() {
    return (
      <React.Fragment>
          <Col sm={6} md={3}>
          <div className="card" style={{ width: '20rem'}} onClick={() => this.props.onClick && this.props.onClick()}>
            <div className="card-body">
                <p className="card-text">{this.props.value}</p>
            </div>
          </div>
          </Col>
      </React.Fragment>
    );
  }
}

export default Card;
