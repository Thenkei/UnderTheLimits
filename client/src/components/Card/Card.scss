//
// Cards style
// --------------------------------------------------

.card {
	display: flex;
	position: relative;
	flex-direction: column;
	-webkit-backface-visibility: hidden;
	backface-visibility: hidden;
	transition: all .235s ease 0s;
	border: 2px solid #e6e7e9;
	box-shadow: 15px 28px 25px -18px rgba( 0, 0, 0, .2 );
	word-wrap: break-word;
	will-change: transform;
}

.card:hover {
	-webkit-transform: translate3d( 0, 2px, 0 );
	transform: translate3d( 0, 2px, 0 );
	box-shadow: 2px 8px 8px -5px rgba( 0, 0, 0, .3 );
}

.card .card-footer,
.card .card-header {
	padding: .75rem 1.25rem;
	background-color: rgba( 0, 0, 0, .03 );
}

.card .card-header {
	border-bottom: 2px solid #e6e7e9;
}

.card .card-footer {
	border-top: 2px solid #e6e7e9;
}

.card .card-body {
	flex: 1 1 auto;
	padding: 1.25rem;
}

.card .card-body .card-title,
.card .card-body h4 {
	margin-top: 0;
	margin-bottom: .5rem;
}

.card .card-body .card-subtitle,
.card .card-body h5 {
	margin-top: 0;
	margin-bottom: .5rem;
	color: #0071de;
}

.card .card-body .card-text,
.card .card-body p {
	margin-top: 0;
	margin-bottom: 1rem;
}

.card .card-body .card-link + .card-link,
.card .card-body a + a {
	margin-left: 1.25rem;
}
