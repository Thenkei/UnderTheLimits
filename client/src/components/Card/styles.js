export default theme => ({
  card: {
    display: 'flex',
    position: 'relative',
    flexDirection: 'column',
    width: '175px',
    minHeight: '230px',
    margin: '15px',
    backfaceVisibility: 'hidden',
    transition: `all ${theme.transitions.easing.easeInOut} ${theme.transitions.duration.short}ms`,
    border: `2px solid ${theme.palette.primary.main}`,
    borderRadius: '8px',
    backgroundColor: 'white',
    boxShadow: '15px 28px 25px -18px rgba( 0, 0, 0, .25)',
    color: theme.palette.primary.main,
    fontSize: '14px',
    wordWrap: 'break-word',
    cursor: 'pointer',
    willChange: 'transform',

    '&:hover': {
      transform: 'translate3d( 0, -6px, 0 )',
      boxShadow: '2px 8px 8px -5px rgba( 0, 0, 0, .5 )',
    },
  },

  cardBody: {
    flex: '1 1 auto',
    padding: theme.spacing.unit * 2,
  },

});
