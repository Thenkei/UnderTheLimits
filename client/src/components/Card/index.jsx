import React from 'react';
import PropTypes from 'prop-types';

import {
  Tooltip,
  Typography,
  withStyles,
} from '..';

import styles from './styles';

const Card = ({
  classes, definition, onClick, checked, value,
}) => {
  let MainWrapper = React.Fragment;
  let wrapperProps = {};
  if (definition && definition.length > 0) {
    MainWrapper = Tooltip;
    wrapperProps = {
      placement: 'top',
      id: definition.toLowerCase().replace(' ', '_'), // For accessibility
      title: definition,
    };
  }

  return (
    <MainWrapper {...wrapperProps}>
      <div
        className={classes.card}
        onClick={() => onClick && onClick()}
        role='none'
        style={{ border: checked && '3px solid red' }}
      >
        <div className='card-body'>
          <Typography className='card-text'>{value}</Typography>
        </div>
      </div>
    </MainWrapper>
  );
};

Card.defaultProps = {
  definition: null,
  onClick: null,
  checked: false,
  value: '',
};

Card.propTypes = {
  classes: PropTypes.object.isRequired,

  definition: PropTypes.string,
  onClick: PropTypes.func,
  checked: PropTypes.bool,
  value: PropTypes.string,
};

export default withStyles(styles, { withTheme: true })(Card);
