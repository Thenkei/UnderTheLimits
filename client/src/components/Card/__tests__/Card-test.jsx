import React from 'react';
import renderer from 'react-test-renderer';

import Card from '..';

it('renders correctly with a few props', () => {
  const tree = renderer.create(
    <Card
      value="BLBL - C'est un test !"
      checked
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
