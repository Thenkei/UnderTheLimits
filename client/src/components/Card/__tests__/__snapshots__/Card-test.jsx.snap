// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with a few props 1`] = `
<div
  className="Card-card-1"
  onClick={[Function]}
  role="none"
  style={
    Object {
      "border": "3px solid red",
    }
  }
>
  <div>
    <p
      className="MuiTypography-root-2 MuiTypography-body1-11 card-text"
    >
      BLBL - C'est un test !
    </p>
  </div>
</div>
`;
